Add sample for custom marshalling in p/invoke source generation (#5315)
dotnet/samples: core/interop/source-generation/custom-marshalling/README.md

## Prerequisites

- [.NET 7 SDK](https://dotnet.microsoft.com/download) Preview 7 or later

- C++ compiler
  - Windows: `cl.exe`
    - See [installation instructions](https://docs.microsoft.com/cpp/build/building-on-the-command-line#download-and-install-the-tools).
  - Linux/macOS: `g++`

## Build and Run

1) In order to build and run, all prerequisites must be installed. The following are also required:

    - On Linux/macOS, the C++ compiler (`g++`) must be on the path.
    - The C++ compiler (`cl.exe` or `g++`) and `dotnet` must be the same bitness (32-bit versus 64-bit).
      - On Windows, the sample is set up to use the bitness of `dotnet` to find the corresponding `cl.exe`

1) Navigate to the root directory and run `dotnet build`

1) Run the samples. Do one of the following:

    - Use `dotnet run` (which will build and run at the same time).
    - Use `dotnet build` to build the executable. The executable will be in `bin` under a subdirectory for the configuration (`Debug` is the default).
        - Windows: `bin\Debug\custommarshalling.exe`
        - Non-Windows: `bin/Debug/custommarshalling`

> Note: The way the sample is built is relatively complicated. The goal is that it's possible to build and run the sample with simple `dotnet run` with minimal requirements on pre-installed tools. Typically real-world projects which have both managed and native components will use different build systems for each; for example msbuild/dotnet for managed and CMake for native.

Reviewer: ```suggestion
Note: The way the sample is built is relatively complicated. The goal is that it's possible to build and run the sample with a simple `dotnet run` command with minimal requirements for pre-installed tools. Typically, real-world projects that have both managed and native components will use different build systems for each; for example, msbuild/dotnet for managed and CMake for native.
```
